### PR TITLE
Remove all global state from setProperty

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Remove all global state in from setProperty
Added properties to ActorClient
Fixed constructor visibility for ActorClient

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: N/A

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: sdk-tests/src/test/java/io/dapr/it/DaprPorts.java [hunk 1]
```diff
@@ -30,10 +34,18 @@ public class DaprPorts {
 
   private final Integer appPort;
 
+  private final Map<Property<?>, String> overrides;
+
   private DaprPorts(Integer appPort, Integer httpPort, Integer grpcPort) {
     this.grpcPort = grpcPort;
     this.httpPort = httpPort;
     this.appPort = appPort;
```
Comment (artur-ciocanu) [inline sdk-tests/src/test/java/io/dapr/it/DaprPorts.java:42]: We are using Java 11, so there is no need to use old Java patterns like `Collections.unmodifiableMap(...)` you could replace all that with:
```
this.overrides = Map.of(
      Properties.GRPC_PORT, grpcPort.toString(),
      Properties.HTTP_PORT, httpPort.toString(),
      Properties.HTTP_ENDPOINT, "http://127.0.0.1:" + httpPort,
      Properties.GRPC_ENDPOINT, "127.0.0.1:" + grpcPort
    );
```
Which I think is more readable and `Map.of()` creates an immutable map behind the scenes.

File: sdk/src/main/java/io/dapr/config/Properties.java [hunk 2]
```diff
@@ -189,10 +191,17 @@ public Properties() {
 
   /**
    * Creates a new instance to handle Properties per instance.
-   * @param overrides to override static properties
-   */
-  public Properties(Map<String, String> overrides) {
-    this.overrides = Collections.unmodifiableMap(overrides);
+   * @param overridesInput to override static properties
+   */
+  public Properties(Map<?, String> overridesInput) {
```
Comment (artur-ciocanu) [inline sdk/src/main/java/io/dapr/config/Properties.java:196]: Here we can also use `Map.of()` and `Map.copyOf()` so the expression below becomes:
```
this.overrides = overridesInput == null ? Map.of() :
        Map.copyOf(overridesInput.entrySet().stream()
            .filter(e -> e.getKey() != null)
            .filter(e -> e.getValue() != null)
            .collect(Collectors.toMap(
                entry -> entry.getKey().toString(),
                entry -> entry.getValue()
            )));
```

File: sdk/src/test/java/io/dapr/utils/NetworkUtilsTest.java [hunk 3]
```diff
@@ -138,17 +144,17 @@ private static void testGrpcEndpointParsingScenario(
       String expectedEndpoint,
       boolean expectSecure
   ) {
```
Comment (artur-ciocanu) [inline sdk/src/test/java/io/dapr/utils/NetworkUtilsTest.java:146]: This expression can be replaced with:
```
var override = Map.of(Properties.GRPC_ENDPOINT.getName(), grpcEndpointEnvValue);
```

File: sdk/src/test/java/io/dapr/utils/NetworkUtilsTest.java [hunk 4]
```diff
@@ -138,17 +144,17 @@ private static void testGrpcEndpointParsingScenario(
...
-    var settings = NetworkUtils.GrpcEndpointSettings.parse(new Properties());
+    var override = Collections.singletonMap(Properties.GRPC_ENDPOINT.getName(), grpcEndpointEnvValue);
+    var settings = NetworkUtils.GrpcEndpointSettings.parse(new Properties(override));
 
     Assertions.assertEquals(expectedEndpoint, settings.endpoint);
     Assertions.assertEquals(expectSecure, settings.secure);
   }
 
   private static void testGrpcEndpointParsingErrorScenario(String grpcEndpointEnvValue) {
     try {
```
Comment (artur-ciocanu) [inline sdk/src/test/java/io/dapr/utils/NetworkUtilsTest.java:155]: This expression can be replaced with:
```
var override = Map.of(Properties.GRPC_ENDPOINT.getName(), grpcEndpointEnvValue);
```

**Review by artur-ciocanu — APPROVED** [review]
@artursouza the PR looks great! I have left a few tiny comments regarding switching to use `Map.of()` instead of relying on `Collections.unmodifiableMap()`.

**Comment by artur-ciocanu** [comment]
@artursouza you might want to re-trigger the build. It seems that there is race condition between Testcontainers and Spring Boot Test Context for a particular test.

**Comment by artur-ciocanu** [comment]
@cicoyle we need your approval again since the PR has changed. Thank you!

**Comment by artur-ciocanu** [comment]
@artursouza it's not critical, but there a few more places in the PR where I mentioned that we could use `Map.of()`.

File: sdk-tests/src/test/java/io/dapr/it/spring/messaging/DaprSpringMessagingIT.java [hunk 5]
```diff
@@ -86,6 +87,7 @@ public void beforeEach() throws InterruptedException {
   }
 
   @Test
+  @Disabled("Test is flaky due to global state in the spring test application.")
```
Comment (artursouza) [inline sdk-tests/src/test/java/io/dapr/it/spring/messaging/DaprSpringMessagingIT.java:90]: @salaboy This test is flaky, it has global state for setting properties somewhere. I disabled it until we find it. Can you help in a new PR after this?

**Comment by artursouza** [comment]
> @artursouza it's not critical, but there a few more places in the PR where I mentioned that we could use `Map.of()`.

Done

File: sdk-tests/src/test/java/io/dapr/it/spring/messaging/DaprSpringMessagingIT.java (hunk 5 above)
Comment (artursouza) [reply sdk-tests/src/test/java/io/dapr/it/spring/messaging/DaprSpringMessagingIT.java:90]: Created a tracking issue.

**Review by cicoyle — DISMISSED** [review]
lgtm

**Review by cicoyle — DISMISSED** [review]

**Review by cicoyle — APPROVED** [review]

**Pull request merged**